common/views/components/IIIFViewer/parts/MainViewer.js

```diff
@@ -0,0 +1,244 @@
+// @flow
+import { memo, useState, useRef } from 'react';
+import { FixedSizeList, areEqual } from 'react-window';
+import debounce from 'lodash.debounce';
+import styled from 'styled-components';
+import LL from '@weco/common/views/components/styled/LL';
+import useScrollVelocity from '@weco/common/hooks/useScrollVelocity';
+import {
+  iiifImageTemplate,
+  convertIiifUriToInfoUri,
+} from '@weco/common/utils/convert-image-uri';
+import ImageViewer from '@weco/common/views/components/ImageViewer/ImageViewer';
+import IIIFResponsiveImage from '@weco/common/views/components/IIIFResponsiveImage/IIIFResponsiveImage';
+import { getCanvasOcr } from '@weco/catalogue/services/catalogue/works';
+
+const ThumbnailWrapper = styled.div`
+  opacity: ${props => (props.imageLoaded ? 1 : 0)};
+  transition: opacity 500ms ease;
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  img {
+    position: relative;
+    display: block;
+    margin: auto;
+    top: 50%;
+    transform: translateY(-50%);
+    max-width: 95%;
+    max-height: 95%;
+    width: auto;
+    height: inherit;
+  }
+`;
+
+function getImageTypeForScrollVelocity(velocity) {
```

Comment:
This could probably either be inline, or more like

```
function getImageTypeForScrollVelocity(velocity) {
  return velocity > 1 ? 'thumbnail' : 'main';
}
```